docs/reference/setup/logging-config.asciidoc

```diff
@@ -194,3 +197,38 @@ files (four rolled logs, and the active log).
 
 You can disable it in the `config/log4j2.properties` file by setting the deprecation
 log level to `error`.
+
+
+[float]
+[[json-logging]]
+=== JSON log format
+
+To make parsing Elasticsearch logs easier, logs are now printed in a JSON format.
+This is configured by a Log4J layout property `appender.rolling.layout.type = ESJsonLayout`.
+This layout requires a `type_name` attribute to be set which is used to distinguish
+logs streams when parsing.
+[source,properties]
+--------------------------------------------------
+appender.rolling.layout.type = ESJsonLayout
+appender.rolling.layout.type_name = server
+--------------------------------------------------
+:es-json-layout-java-doc: {elasticsearch-javadoc}/org/elasticsearch/common/logging/ESJsonLayout.html
+
+Each line contains a single JSON document with the properties configured in `ESJsonLayout`.
+See this class {es-json-layout-java-doc}[javadoc] for more details.
+However if a JSON document contains an exception, it will be printed over multiple lines.
+The first line will contain regular properties and subsequent lines will contain the
+stacktrace formatted as a JSON array.
+
+
+NOTE: You can still use your own custom layout. To do that replace the line
```

Comment:
I think the docs need to take into account that we now log as JSON *and* in plain text.